### PR TITLE
feat(plugin): sticky columns

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Two major things in this PR:

- Completes the StickyColumns Plugin
  - implementation
  - docs
  - tests
- Managing of "styles" in the built-in plugins is now done via a styles object, `CSSStyleDeclarations`, rather than a string.
  - This is so that individual styles can be merged by many plugins, whereas before `style` setting would override other plugin's work (for example, resizing would override sticky's styles)
  
**Test Plan**
  
- visit: https://e8f437bd.ember-headless-table.pages.dev/docs/plugins/sticky-column
- mess around with the demo
- read the docs

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -75,7 +75,8 @@ jobs:
           - typescript@4.5
           - typescript@4.6
           - typescript@4.7
-          - typescript@4.8
+          # https://github.com/typed-ember/glint/issues/406
+          # - typescript@4.8
```
Comment (NullVoxPopuli) [inline .github/workflows/ci.yml:79]: clickable: https://github.com/typed-ember/glint/issues/406

**Comment by nicolechung** [comment]
Screen Shot 2022-10-12 at 8 55 05 AM

The "Edit this demo" link currently 404's in the do-it-live link, is this intentional? Like, does it only work in dev mode?

**Review by nicolechung — APPROVED** [review]
A very minor nit: I am not clear by clicking on the demo on what sticky:left/sticky:right does since A is sticking left and G is sticking right.

Maybe if "F" was also sticking left I could see the difference?

**Comment by NullVoxPopuli** [comment]
Does the code help? 
```
const leftSticky = () => StickyColumns.forColumn(() => ({ sticky: 'left' }));
const rightSticky = () => StickyColumns.forColumn(() => ({ sticky: 'right' }));

export default class extends Component {
  table = headlessTable(this, {
    columns: () => [
      { name: 'column A', key: 'A', pluginOptions: [leftSticky(), ...] },
     // ...
      { name: 'column G', key: 'G', pluginOptions: [rightSticky(), ...] },
```

**Comment by nicolechung** [comment]
>But in general I want to make all these demos have configurable options -- but I need to invent a UI for that 😅

Ah gotcha. Yeah I def don't think it's a blocker and I can play around with it on localhost in my own time to understand it.

**Review by nicolechung — CHANGES_REQUESTED** [review]
Question about the `sticky` style attribute instead of a sticky class.

**Review by nicolechung — APPROVED** [review]

**Pull request merged**